tests/components/overkiz/test_init.py

```diff
@@ -0,0 +1,94 @@
+"""Tests for Overkiz integration init."""
+from homeassistant.components.overkiz.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.setup import async_setup_component
+
+from .test_config_flow import TEST_EMAIL, TEST_GATEWAY_ID, TEST_HUB, TEST_PASSWORD
+
+from tests.common import MockConfigEntry, mock_registry
+
+ENTITY_SENSOR_DISCRETE_RSSI_LEVEL = "sensor.zipscreen_woonkamer_discrete_rssi_level"
+ENTITY_ALARM_CONTROL_PANEL = "alarm_control_panel.alarm"
+ENTITY_SWITCH_GARAGE = "switch.garage"
+ENTITY_SENSOR_TARGET_CLOSURE_STATE = "sensor.zipscreen_woonkamer_target_closure_state"
+ENTITY_SENSOR_TARGET_CLOSURE_STATE_2 = (
+    "sensor.zipscreen_woonkamer_target_closure_state_2"
+)
+
+
+async def test_unique_id_migration(hass: HomeAssistant) -> None:
+    """Test migration of sensor unique IDs."""
+
+    mock_entry = MockConfigEntry(
+        domain=DOMAIN,
+        unique_id=TEST_GATEWAY_ID,
+        data={"username": TEST_EMAIL, "password": TEST_PASSWORD, "hub": TEST_HUB},
+    )
+
+    mock_entry.add_to_hass(hass)
+
+    mock_registry(
+        hass,
+        {
+            ENTITY_SENSOR_DISCRETE_RSSI_LEVEL: er.RegistryEntry(
+                entity_id=ENTITY_SENSOR_DISCRETE_RSSI_LEVEL,
+                unique_id="io://1234-5678-1234/3541212-OverkizState.CORE_DISCRETE_RSSI_LEVEL",
+                platform=DOMAIN,
+                config_entry_id=mock_entry.entry_id,
+            ),
+            ENTITY_ALARM_CONTROL_PANEL: er.RegistryEntry(
+                entity_id=ENTITY_ALARM_CONTROL_PANEL,
+                unique_id="internal://1234-5678-1234/alarm/0-UIWidget.TSKALARM_CONTROLLER",
+                platform=DOMAIN,
+                config_entry_id=mock_entry.entry_id,
+            ),
+            ENTITY_SWITCH_GARAGE: er.RegistryEntry(
+                entity_id=ENTITY_SWITCH_GARAGE,
+                unique_id="io://1234-5678-1234/0-UIClass.ON_OFF",
+                platform=DOMAIN,
+                config_entry_id=mock_entry.entry_id,
+            ),
+            ENTITY_SENSOR_TARGET_CLOSURE_STATE: er.RegistryEntry(
+                entity_id=ENTITY_SENSOR_TARGET_CLOSURE_STATE,
+                unique_id="io://xxxx-xxxx-xxxx/3541212-OverkizState.CORE_TARGET_CLOSURE",
+                platform=DOMAIN,
+                config_entry_id=mock_entry.entry_id,
+            ),
+            ENTITY_SENSOR_TARGET_CLOSURE_STATE_2: er.RegistryEntry(
+                entity_id=ENTITY_SENSOR_TARGET_CLOSURE_STATE_2,
+                unique_id="io://xxxx-xxxx-xxxx/3541212-core:TargetClosureState",
+                platform=DOMAIN,
+                config_entry_id=mock_entry.entry_id,
+            ),
```

Comment:
To make the test easier to follow, add some comments to this dictionary, for example:
To make the test easier to follow, add some comments to this dictionary, for example:
```py
           # This will be migrated to "io://1234-5678-1234/0-OnOff"
            ENTITY_SWITCH_GARAGE: er.RegistryEntry(
                entity_id=ENTITY_SWITCH_GARAGE,
                unique_id="io://1234-5678-1234/0-UIClass.ON_OFF",
                platform=DOMAIN,
                config_entry_id=mock_entry.entry_id,
            ),
            # This will be deleted because an entity with unique ID
            # "io://xxxx-xxxx-xxxx/3541212-core:TargetClosureState" already exists
            ENTITY_SENSOR_TARGET_CLOSURE_STATE: er.RegistryEntry(
                entity_id=ENTITY_SENSOR_TARGET_CLOSURE_STATE,
                unique_id="io://xxxx-xxxx-xxxx/3541212-OverkizState.CORE_TARGET_CLOSURE",
                platform=DOMAIN,
                config_entry_id=mock_entry.entry_id,
            ),
            # This will not be migrated
            ENTITY_SENSOR_TARGET_CLOSURE_STATE_2: er.RegistryEntry(
                entity_id=ENTITY_SENSOR_TARGET_CLOSURE_STATE_2,
                unique_id="io://xxxx-xxxx-xxxx/3541212-core:TargetClosureState",
                platform=DOMAIN,
                config_entry_id=mock_entry.entry_id,
            ),
```

Reply:
Thanks. Fixed in 849be3bd45b7dc022df0b6dc77ff6ecb63007f03.